src/services/UserService.js

```diff
@@ -0,0 +1,90 @@
+import { $db } from '@/services/firebase'
+import { configService } from '@/services/Config.service.js'
+
+const COLLECTION_NAME = 'user'
+
+class UserService {
+  constructor () {
+    this.user = null
+  }
+
+  async signIn ({ firstName, lastName, initials, masterPassword }) {
+    const loadedPassword = await configService
+      .fetchMasterPassword()
+    if (loadedPassword !== masterPassword) {
+      return true
+    } else {
+      await this.fetchUserByInitials(initials)
+        .then(user => {
+          if (user) {
+            return user
+          } else {
+            const user1 = {
+              firstName,
+              lastName,
+              initials
+            }
+            return $db().collection(COLLECTION_NAME).add(user1).then(response => {
+              user1.id = response.id
+              return user1
+            })
+          }
+        })
+        .then(user2 => {
+          this.user = user2
+          localStorage.userId = user2.id
+        })
+    }
+  }
+
+  async fetchUserByInitials (initials) {
+    const response = await $db().collection(COLLECTION_NAME).where('initials', '==', initials).get()
+    let user = null
+    response.forEach(userDoc => {
+      user = {
+        id: userDoc.id,
+        firstName: userDoc.data().firstName,
+        lastName: userDoc.data().lastName,
+        initials: userDoc.data().initials
+      }
+    })
```

Comment:
Hier könnte man das Object on the fly erstellen.
```suggestion
    response.forEach(userDoc => {
     const user = {
        id: userDoc.id,
        firstName: userDoc.data().firstName,
        lastName: userDoc.data().lastName,
        initials: userDoc.data().initials
      }
    })
```